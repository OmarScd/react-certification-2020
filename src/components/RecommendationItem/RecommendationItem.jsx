import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

const Container = styled(Link)`
  color: ${(props) => props.theme.colors.font};
  border: 1px solid black;
  display: flex;
  text-decoration: none;
`;

const Thumbnail = styled.img`
  margin: 0;
  width: 120px;
`;

const Title = styled.h3`
  margin: 0;
  overflow: hidden;
  padding-left: 2px;
`;

const RecommendationItem = ({ id, thumbnail, title }) => {
  return (
    <Container to={`/${id}`}>
      <Thumbnail src={thumbnail} />
      <Title>{title}</Title>
    </Container>
  );
};

export default RecommendationItem;
