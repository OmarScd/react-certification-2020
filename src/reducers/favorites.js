const favoritesReducer = (state, action) => {
  switch (action.type) {
    case 'POPULATE_FAVORITES':
      return action.favorites;
    case 'ADD_FAVORITE':
      return [...state, action.favorite];
    case 'REMOVE_FAVORITE':
      return state.filter((favorite) => favorite.id !== action.id);
    case 'CLEAR_FAVORITES':
      return [];
    default:
      return state;
  }
};

export default favoritesReducer;
