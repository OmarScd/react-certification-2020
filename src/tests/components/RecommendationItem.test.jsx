import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import RecommendationItem from '../../components/RecommendationItem';
import favorites from '../fixtures/favorites';
import ThemeProvider from '../../providers/Theme';

describe('RecommendationItem', () => {
  afterAll(cleanup);

  it('renders correctly', () => {
    const recommendation = favorites[0];
    const { container } = render(
      <ThemeProvider>
        <BrowserRouter>
          <RecommendationItem
            id={recommendation.id}
            thumbnail={recommendation.thumbnail}
            title={recommendation.title}
          />
        </BrowserRouter>
      </ThemeProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
