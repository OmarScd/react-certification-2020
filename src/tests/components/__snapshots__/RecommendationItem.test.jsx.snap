// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecommendationItem renders correctly 1`] = `
<a
  class="sc-bdnylx djbTLE"
  href="/video/YNZmZ4ARr38"
>
  <img
    class="sc-gtssRu jwjGxM"
    src="https://i.ytimg.com/vi/YNZmZ4ARr38/mqdefault.jpg"
  />
  <h3
    class="sc-dlnjPT egPkLV"
  >
    Shrek Forever After (2010) - Daddy Ever After Scene (2/10) | Movieclips
  </h3>
</a>
`;
